app/utils/IncomeSourcesUtils.scala

```diff
@@ -42,39 +42,16 @@ trait IncomeSourcesUtils extends FeatureSwitching {
     }
   }
 
-  def newWithIncomeSourcesRemovedFromSession(redirect: Result, sessionService: SessionService, errorRedirect: Result)(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+  def withIncomeSourcesRemovedFromSession(redirect: Result)
+                                         (implicit user: MtdItUser[_],
+                                          sessionService: SessionService,
+                                          ec: ExecutionContext
+                                         ): Future[Result] = {
     sessionService.remove(SessionKeys.incomeSourcesSessionKeys, redirect).map {
-      case Left(_) => errorRedirect
+      case Left(_) => throw new Exception("Failed to remove income sources from session")
```

Comment:
can you add details of what key we failed to remove in the exception message ?

Reply:
not sure if you understand this correctly, you can add failed key in the Exception message